### PR TITLE
Add config for vsphere upi compact mode deployments

### PR DESCRIPTION
**Pull request by clacroix12**
Issue: https://url.corp.redhat.com/eb865d5

### CODE REVIEW

**Review by shylesh — DISMISSED** [review]
+1

**Review by petr-balogh — DISMISSED** [review]

**Review by vavuthu — APPROVED** [review]

File: conf/deployment/vsphere/upi_1az_rhcos_vsan_compact_mode_3m_0w.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+DEPLOYMENT:
+  allow_lower_instance_requirements: false
+ENV_DATA:
+  platform: 'vsphere'
+  deployment_type: 'upi'
+  worker_replicas: 0
+  master_replicas: 3
+  worker_num_cpus: '16'
+  master_num_cpus: '4'
+  master_memory: '16384'
```
Comment (prsurve) [inline conf/deployment/vsphere/upi_1az_rhcos_vsan_compact_mode_3m_0w.yaml:12]: with 4 CPU and 16gb ram, i don't think that will be sufficient for running OCS on them

Comment (clacroix12) [reply conf/deployment/vsphere/upi_1az_rhcos_vsan_compact_mode_3m_0w.yaml]: I've updated the values for master nodes to reflect what our worker nodes are using in standard deployments and removed the values for workers as we don't have any worker nodes anyways. @petr-balogh FYI.

**Comment by petr-balogh** [comment]
Hello @clacroix12 , have you progressed with the verification of this PR to see what else we need to change to support this combination in vSphere?

File: ocs_ci/deployment/vmware.py [hunk 2]
```diff
@@ -891,6 +891,21 @@ def generate_manifests(self):
...
+                )
+
+                with open(cluster_scheduler_config, "r") as f:
+                    data = yaml.safe_load(f)
+
+                data["spec"]["mastersSchedulable"] = True
+
+                with open(cluster_scheduler_config, "w") as f:
+                    yaml.dump(data, f)
+
```
Comment (prsurve) [inline ocs_ci/deployment/vmware.py:908]: AFAIK if we are setting worker count as 0 in install-config.yaml we don't need to make any changes to make master node schedulable

**Comment by clacroix12** [comment]
> Hello @clacroix12 , have you progressed with the verification of this PR to see what else we need to change to support this combination in vSphere?

Hey Petr, just following up here for clarity. I'm currently seeing issues with the ingress controller, I believe this has to do with one of the steps mentioned in the [doc](https://docs.openshift.com/container-platform/4.13/installing/installing_vsphere/installing-vsphere-three-node.html#installing-vsphere-three-node) about configuring traffic to go to the control plane nodes instead of the worker nodes.

I'm currently trying to find a solution to configure the load balancer to direct traffic as required.

The error I'm seeing with regards to the ingress controller: 
```
"The \"default\" ingress controller reports Degraded=True: DegradedConditions: One or more other status conditions indicate a degraded state: DeploymentAvailable=False (DeploymentUnavailable: The deployment has Available status
condition set to False (reason: MinimumReplicasUnavailable) with message: Deployment does not have minimum availability.), DeploymentReplicasMinAvailable=False (DeploymentMinimumReplicasNotMet: 0/2 of replicas are available, max unavailable is 1: Some pods are not scheduled: Pod \"router-default-58f74c5b6f-v8k6s\" cannot be scheduled: 0/3 nodes are available: 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling.. Pod \"router-default-58f74c5b6f-whkqc\" cannot be scheduled: 0/3 nodes are available: 3 node(s) had untolerated taint {node-role.kubernetes.io/master: }. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling.. Make sure you have sufficient worker nodes.), DeploymentReplicasAllAvailable=False (DeploymentReplicasNotAvailable: 0/2 of replicas are available), CanaryChecksSucceeding=Unknown (CanaryRouteNotAdmitted: Canary route is not admitted by the default ingress controller)"
```

**Comment by clacroix12** [comment]
Testing with UPI has been unsuccessful, however IPI deployments are working with the compact configuration. Closing this PR and will open another with the IPI config only.

**Pull request closed without merging**